lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestContextStart.cs

```diff
@@ -0,0 +1,201 @@
+                    .Set(TaskConfiguration.Identifier, "TaskID")
+                    .Set(TaskConfiguration.Task, GenericType<TestTask>.Class)
+                    .Build();
+                value.SubmitTask(c);
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID")
+                        .Set(ContextConfiguration.OnContextStart, GenericType<ContextStartHandler>.Class)
+                        .Build());
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class ContextStartHandler : IObserver<IContextStart>
+        {
+            private readonly DataDownLoader _dataDownLoader;
+
+            [Inject]
+            private ContextStartHandler(DataDownLoader dataDownLoader)
+            {
+                _dataDownLoader = dataDownLoader;
+            }
+
+            public void OnNext(IContextStart value)
+            {
+                Logger.Log(Level.Info, StartedHandlerMessage);
+                _dataDownLoader.LoadData();
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class DataDownLoader
+        {
+            private bool _dataLoaded;
+
+            [Inject]
+            private DataDownLoader()
+            {
+                _dataLoaded = false;
+            }
+
+            public void LoadData()
+            {
+                Logger.Log(Level.Info, DataDownLoadStartedMessage);
+                _dataLoaded = true;
+            }
+
+            public bool DataLoaded
+            {
+                get { return _dataLoaded; }
+            }
+        }
+
+        private sealed class TestTask : ITask
+        {
+            private readonly DataDownLoader _dataDownLoader;
+
+            [Inject]
+            private TestTask(DataDownLoader dataDownLoader)
+            {
+                _dataDownLoader = dataDownLoader;
+            }
+
+            public void Dispose()
+            {
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                Logger.Log(Level.Info, "Hello in TestTask");
+                if (_dataDownLoader.DataLoaded == true)
+                {
+                    Logger.Log(Level.Info, DataDownLoadCompletedMessage);
+                    return null;
+                }
+                Thread.Sleep(5 * 1000);
```

Comment:
We usually sleep little bit to simulate task doing something. But not necessary in this case.